### PR TITLE
Fixing templates for java model generated files import statement for …

### PR DESCRIPTION
**Pull request by deepak2016**
…models related files.

This is an initial change to fix imports for java files. (Formatting appearing in visual studio and in github change looks a bit different)
These changes in templates only fixes the import statements and doesn't change any other content of the java files.

Testing steps:
1. Generated models and requests files using generator code.
2. Copied over models-generated and requests-generated completely to java sdk corresponding directory
3. Copied over new models-extensions and requests-extensions file to java sdk corresponding locations
4. Built java sdk with success

### CODE REVIEW

**Comment by MIchaelMainer** [comment]
I'd like to add a step 1A to the review for change steps:
1A) Compared the files generated with the changes against existing files and see the expected changes in existing files.

File: Templates/Java/BaseJavaModel.template.tt [hunk 1]
```diff
@@ -506,6 +506,148 @@
...
+					sb.Append("\n");
+		}
+		if(baseTypeNameStr == "BasePlannerChecklistItems")
+		{
+			sb.AppendFormat(format,
+						host.CurrentModel.NamespaceName(),
+						"models.extensions",
+						 "PlannerChecklistItem");
+					sb.Append("\n");
+		}
```
[the remark this answers is not in the document]
Comment (deepak2016) [reply Templates/Java/BaseJavaModel.template.tt]: These are due to overrides present in CustomOverwrites.cs file. Needed to handle these cases separately.

File: Templates/Java/BaseJavaModel.template.tt [hunk 2]
```diff
@@ -506,6 +506,148 @@
...
+		 sb.AppendFormat(@"import {0}.concurrency.*;
+import {0}.core.*;
+import {0}.http.*;
+import {0}.options.*;
+import {0}.serializer.*;
+import java.util.Arrays;
+import java.util.EnumSet;", host.CurrentModel.NamespaceName());
+
+		sb.Append("\n");		
+        var format = @"import {0}.{1}.{2};";
```
[the remark this answers is not in the document]
Comment (deepak2016) [reply Templates/Java/BaseJavaModel.template.tt]: Renamed.

File: Templates/Java/BaseJavaModel.template.tt [hunk 3]
```diff
@@ -506,6 +506,148 @@
...
+						"models.extensions",
+						 "PlannerChecklistItem");
+					sb.Append("\n");
+		}
+		
+		if (properties != null)
+        {
+            foreach (var property in properties.Where(p => p.IsCollection() && p.IsNavigation())) 
+			{
+				var propertyType = BaseTypeCollectionResponse(property);
```
[the remark this answers is not in the document]
Comment (deepak2016) [reply Templates/Java/BaseJavaModel.template.tt]: Moved.

File: Templates/Java/BaseJavaModel.template.tt [hunk 4]
```diff
@@ -506,6 +506,148 @@
         }
         return "<" + BaseTypeCollectionResponse(c) + ", " + ITypeCollectionPage(c) + ">";
     }
+
```
[the remark this answers is not in the document]
Comment (deepak2016) [reply Templates/Java/BaseJavaModel.template.tt]: Added comments

**Pull request merged**